File: Data Structures & Algorithms/maximum-subarray/submission-5.java
class Solution {
    public int maxSubArray(int[] nums) {
        int currBest = nums[0];
        int currSum = 0;
        for (int i = 0; i < nums.length; i++) {
            currSum = Math.max(currSum, 0);

            currSum += nums[i];
            currBest = Math.max(currSum, currBest);
        }

        return currBest;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-5.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l, r = 1, max(piles)
        res = 0

        while l <= r:
            mid = l + (r- l) // 2
            
            time = 0
            for p in piles:
                time += math.ceil(float(p) / mid)
            if time <= h:
                r = mid - 1
                res = mid
            else:
                l = mid + 1
        
        return res